src/marqo/core/index_management/index_management.py

```diff
@@ -115,63 +118,55 @@ def delete_index(self, marqo_index: MarqoIndex) -> None:
         """
         Delete a Marqo index.
 
+        This method is idempotent and does not raise an error if the index does not exist.
+
         Args:
             marqo_index: Marqo index to delete
         """
-        self.delete_index_by_name(marqo_index.name)
+        app = self.vespa_client.download_application()
+
+        self._remove_schema(app, marqo_index.schema_name)
+        self._remove_schema_from_services(app, marqo_index.schema_name)
+        self._add_schema_removal_override(app)
+        self.vespa_client.deploy_application(app)
+        self._delete_index_settings_by_name(marqo_index.name)
```

Comment:
these look like a lot of steps that depend on unreliable hardware (network) - are there adequate safeguards in case any step fails?

Reply:
The process is idempotent. Failure of intermediate steps can create an inconsistent state (e.g., Vespa schema doesn't exist, but Marqo thinks the index exists), but this can be recovered from by calling the API again and getting a 200 response.

Reply:
do the errors the users get descriptive enough for them to take the correct action? 

And, in these inconsistent states, is it ever possible to get into a catastrophic unrecoverable state? Let's say that another user is doing other odd index management requests concurrently?